### ISSUE TITLE
Named import for any module doesn't work

### ISSUE THREAD

**Issue by kitsonk** [microsoft/TypeScript#42506]
# Bug Report

### 🔎 Search Terms

any module export assignment named import

### 🕗 Version & Regression Information

TypeScript v4.1.2

### 💻 Code

*mock.d.ts*

```ts
declare const _: any;
export = _;
```

*example.ts*

```ts
import { a } from "./mock";

import * as b from "./mock";
const { c } = b;

const { d } = await import("./mock");

console.log(a, c, d);
```

*tsconfig.json*

```json
{
  "compilerOptions": {
    "module": "ESNext",
    "strict": true,
    "target": "ESNext",
  },
  "include": [
    "./**/*.ts"
  ]
}
```

### 🙁 Actual behavior

The first import returns the diagnostic:

```
error TS2305: Module '"./mock"' has no exported member 'a'.
```

While `c` and `d` do not error.

### 🙂 Expected behavior

With an export being effectively `any`, I would expect default, namespace, dynamic and named imports to all behave as if the module was `any`.  I cannot find any other way to declare a module in a fashion where the compiler treats any import from that module as `any` and does not issue diagnostics.

As implied in the above, there are certain situations where there is a need to "mock" a module where its shape cannot be statically determined at compile time.

**Comment by RyanCavanaugh** [comment]
This is longstanding behavior because we don't know if `a` is a type, value, namespace, or some combination thereof, and we'd get ourselves into a mess trying to reason about whether or not you were then allowed to declare a local with a nonconflicting name.

With `import type` now, maybe we can just pretend that `any`-sourced bindings are just values only.

**Comment by kitsonk** [comment]
Could also the behaviour be changed under `isolatedModules` where TypeScript is forced to assume it is a value?
